Simple setup

pooja-ces opened the pull request.

codecrafter-a committed:
simple project setup

codecrafter-a committed:
added webpack and babel

codecrafter-a committed:
setup code-standard

codecrafter-a commented on src/App.tsx:
```diff
@@ -0,0 +1,11 @@
+import './App.css'
+
+function App() {
+  return (
```
use react.fragement there not <>

pooja-ces replied:
used React.Fragment instead of <>

codecrafter-a committed:
added vitest and types for image

codecrafter-a committed:
Added fragment

codecrafter-a committed:
added common types. input field and proper file and folder structure

codecrafter-a committed:
fixed form context

codecrafter-a committed:
added test cases

codecrafter-a commented on src/components/InputField.tsx:
Can we make the common folder and put there?

codecrafter-a commented on .eslintrc.js:
```diff
@@ -0,0 +1,28 @@
+module.exports = {
```
**1. Code Formatting and Linting**

Missing ESLint Configuration: Ensure the repository has an .eslintrc file with rules configured for React, Next.js, and Jest.
Prettier Integration: Include a .prettierrc file for consistent code formatting across the project.

Lint Staged & Husky: Add lint-staged and husky to run linting and tests automatically on pre-commit.

codecrafter-a commented on .eslintrc.json:
-Modular Folder Structure: Ensure the project has a well-organized structure, e.g.:css


src/
  components/
  pages/
  hooks/
  utils/
  styles/
  tests/

codecrafter-a commented on eslint.config.js:
Avoid Clutter: Check if all files are placed meaningfully and unused or temporary files are excluded.

codecrafter-a commented on src/utils/validation.test.ts:
Strict TypeScript Settings: Enforce strict settings in tsconfig.json (strict, noImplicitAny, strictNullChecks).
Type Annotations: Ensure all functions, components, and props are fully typed.
Global Types: Use a types directory for reusable type definitions.

codecrafter-a commented on src/components/FormContainer.test.tsx:
Missing Coverage Report: Ensure jest.config.js includes configuration for test coverage reporting.
Test File Locations: Test files should mirror the folder structure, e.g., __tests__/componentName.test.js.
Mocking Utilities: Include mocks for APIs, localStorage, and other browser functionalities.
Integration Tests: Add integration test examples, particularly for key components or API calls.

codecrafter-a commented on src/components/FormContainer.test.tsx:
Environment Variables Handling: Use a .env file for sensitive keys and ensure .env.example is included.
Custom Webpack or Babel Configuration: Verify that any custom settings align with the intended use of the boilerplate.

codecrafter-a commented on src/components/InputField.test.tsx:
README.md Updates: The README file should explain:
Project structure
Setup instructions
Running tests
Deployment steps
Code Comments: Key components and utility functions should have comments explaining their purpose.

codecrafter-a commented on src/contexts/FormContext.tsx:
Atomic Design Principles: Components should follow reusable principles (atoms, molecules, organisms).
Prop Validation: Use PropTypes or TypeScript to validate component props.

codecrafter-a commented on src/hooks/useFormContext.ts:
ARIA Attributes: Check for missing ARIA attributes for better accessibility (e.g., aria-label, role).
Accessible Design: Ensure elements like buttons and forms are keyboard navigable.

codecrafter-a commented on src/index.css:
Code Splitting: Use React.lazy and dynamic imports for better performance.
Tree Shaking: Ensure unused code is eliminated via Next.js configuration.
Image Optimization: Leverage the Next.js Image component for optimized image loading.

codecrafter-a commented on src/pages/UserForm.test.tsx:
Dependencies Audit: Run npm audit or yarn audit to check for vulnerabilities.
Sensitive Data: Ensure no sensitive data or keys are hardcoded in the codebase.

codecrafter-a commented on src/pages/UserForm.tsx:
Error Boundaries: Add an ErrorBoundary component to catch React rendering errors.
API Error Handling: Ensure robust error handling for API calls with fallback UI or messages.

codecrafter-a commented on src/styles/index.css:
Theme Consistency: Ensure global themes are defined in Tailwind config or MUI theme provider.
Utility Classes: Avoid overuse of inline styles or redundant utility classes.

codecrafter-a commented on src/types/formTypes.ts:
Static File Optimization: Verify that static assets are being optimized during builds.
Bundle Analysis: Use tools like next-bundle-analyzer to track bundle size.

codecrafter-a committed:
coding standard
- added description in reademe file
- added husky
 - changed prettierrc file
 - setup tailwindcss
 - added lazy loading
